firebase-fetcher/src/main/java/com/ackee/versionupdatehandler/FirebaseVersionFetcher.kt

```diff
@@ -0,0 +1,42 @@
+package com.ackee.versionupdatehandler
+
+import com.ackee.versioupdatehandler.VersionFetcher
+import com.ackee.versioupdatehandler.model.BasicVersionsConfiguration
+import com.ackee.versioupdatehandler.model.VersionsConfiguration
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Class that fetches version configuration from Firebase Remote.
+ */
+class FirebaseVersionFetcher constructor(
+    private val cacheExpiration: Int = 3600,
+    private val minimalAttributeName: String = MINIMAL_VERSION,
+    private val currentAttributeName: String = CURRENT_VERSION
+) : VersionFetcher {
+
+    override suspend fun fetch(): VersionsConfiguration {
+        val remoteConfig = FirebaseRemoteConfig.getInstance().apply {
+            setDefaultsAsync(DEFAULTS)
+        }
+        val isDevMode = FirebaseRemoteConfig.getInstance().info.configSettings.isDeveloperModeEnabled
+        remoteConfig.fetch((if (isDevMode) 0 else cacheExpiration).toLong()).await()
+        remoteConfig.activate().await()
+
+        val minimalVersion = remoteConfig.getLong(minimalAttributeName)
+        val currentVersion = remoteConfig.getLong(currentAttributeName)
+        return BasicVersionsConfiguration(minimalVersion, currentVersion)
+    }
+
+    companion object {
```

Comment:
we define companions at the top

Reply:
07d724d0eabe5f4ccb1d447515e4b174eef4fdf0